### ISSUE TITLE
Attempted to update component `CalendarList` that has already been unmounted (or failed to mount).

### ISSUE THREAD

**Issue by joseguya** [wix/react-native-calendars#168]
## Description
I was trying the example provided for the Agenda component. After copying the code, exactly as showed in the example: 

`import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
  View,
  Text,
  StyleSheet
} from 'react-native';
import { Header } from 'react-native-elements';
import { NavigationActions } from 'react-navigation';
import { Agenda, LocaleConfig, Calendar } from 'react-native-calendars';
import { logOutUser } from '../actions';
import { ContainersStyles, TextStyles } from '../styles/';

const resetAction = NavigationActions.reset({
    index: 0,
    key: null,
    actions: [NavigationActions.navigate({ routeName: 'Auth' })]
  });

  LocaleConfig.locales['py'] = {
    monthNames: ['Enero', 'Febrero', 'Marzo', 'Abril', 'Mayo', 'Junio', 'Julio', 'Agosto', 'Septiembre', 'Octubre', 'Noviembre', 'Diciembre'],
    monthNamesShort: ['Ene.', 'Feb.', 'Mar.', 'Abr.', 'Mayo', 'Jun.', 'Jul.', 'Ag.', 'Sept.', 'Oct.', 'Nov.', 'Dec.'],
    dayNames: ['Domingo', 'Lunes', 'Martes', 'Miercoles', 'Jueves', 'Viernes', 'Sábado'],
    dayNamesShort: ['Dom.', 'Lun.', 'Mar.', 'Mier.', 'Jue.', 'Vien.', 'Sab.']
  };
class FieldAdminScreen extends Component {
    constructor(props) {
        super(props);
        this.state = {
            items: {}
        };
    }
    
    ComponentWillMount() {
      LocaleConfig.defaultLocale = 'py';      
    }
    logOut() {
        this.props.logOutUser();
        this.props.navigate.dispatch(resetAction);
    }

    loadItems(day) {
        setTimeout(() => {
          for (let i = -15; i < 85; i++) {
            const time = day.timestamp + (i * 24 * 60 * 60 * 1000);
            const strTime = this.timeToString(time);
            if (!this.state.items[strTime]) {
              this.state.items[strTime] = [];
              const numItems = Math.floor(Math.random() * 5);
              for (let j = 0; j < numItems; j++) {
                this.state.items[strTime].push({
                  name: `Item for  + ${strTime}`,
                  height: Math.max(50, Math.floor(Math.random() * 150))
                });
              }
            }
          }
          //console.log(this.state.items);
          const newItems = {};
          Object.keys(this.state.items).forEach(key => { newItems[key] = this.state.items[key]; });
          this.setState({
            items: newItems
          });
        }, 1000);
        // console.log(`Load Items for ${day.year}-${day.month}`);
      }
          
      timeToString(time) {
        const date = new Date(time);
        return date.toISOString().split('T')[0];
      }

      rowHasChanged(r1, r2) {
        return r1.name !== r2.name;
      }

      renderEmptyDate() {
        return (
          <View style={styles.emptyDate}><Text>This is empty date!</Text></View>
        );
      }
    

      renderItem(item) {
        return (
          <View style={[styles.item, { height: item.height }]}><Text>{item.name}</Text></View>
        );
      }

    render() {
        return (
            <View style={ContainersStyles.bookingScreenStyle}>
                <Header 
                    centerComponent={{ text: 'Administrador Pelota Jara', style: TextStyles.bookingHeaderTextStyle }} 
                    outerContainerStyles={ContainersStyles.bookingsHeaderStyle}
                />
                <Agenda
                    items={this.state.items}
                    loadItemsForMonth={this.loadItems.bind(this)}
                    selected={'2017-09-4'}
                    minDate={Date()}
                    renderItem={this.renderItem.bind(this)}
                    renderEmptyDate={this.renderEmptyDate.bind(this)}
                    rowHasChanged={this.rowHasChanged.bind(this)}
                    monthFormat={'MM'}
                    //theme={{ calendarBackground: 'red', agendaKnobColor: 'green' }}
                    //renderDay={(day, item) => (<Text>{day ? day.day: 'item'}</Text>)}
                />
            </View>
        );
    }

}

const mapStateToProps = ({ bookings, auth }) => {
    const { activeBookings, loadingBookings, doneLoadingBookings } = bookings;
    const { user } = auth;
    return { activeBookings, user, loadingBookings, doneLoadingBookings };
};

export default connect(mapStateToProps, {
    logOutUser,
})(FieldAdminScreen);

const styles = StyleSheet.create({
    item: {
      backgroundColor: 'white',
      flex: 1,
      borderRadius: 5,
      padding: 10,
      marginRight: 10,
      marginTop: 17
    },
    emptyDate: {
      height: 15,
      flex: 1,
      paddingTop: 30
    }
  });
`

After running on Expo: 
_

> Attempted to update component `CalendarList` that has already been unmounted (or failed to mount).

_

And with this error: 

```
TypeError: Cannot read property '6' of undefined
    at getTokenReplacement (C:\reactnative\mobile-app-react-native\node_modules\xdate\src\xdate.js:583)
    at processTokenString (C:\reactnative\mobile-app-react-native\node_modules\xdate\src\xdate.js:546)
    at _format (C:\reactnative\mobile-app-react-native\node_modules\xdate\src\xdate.js:524)
    at format (C:\reactnative\mobile-app-react-native\node_modules\xdate\src\xdate.js:513)
    at XDate.proto.toString (C:\reactnative\mobile-app-react-native\node_modules\xdate\src\xdate.js:455)
    at new CalendarList (C:\reactnative\mobile-app-react-native\node_modules\react-native-calendars\src\calendar-list\index.js:45)
    at C:\reactnative\mobile-app-react-native\node_modules\react-native\Libraries\Renderer\ReactNativeStack-dev.js:1694
    at measureLifeCyclePerf (C:\reactnative\mobile-app-react-native\node_modules\react-native\Libraries\Renderer\ReactNativeStack-dev.js:1625)
    at ReactCompositeComponentWrapper._constructComponentWithoutOwner (C:\reactnative\mobile-app-react-native\node_modules\react-native\Libraries\Renderer\ReactNativeStack-dev.js:1693)
    at ReactCompositeComponentWrapper._constructComponent (C:\reactnative\mobile-app-react-native\node_modules\react-native\Libraries\Renderer\ReactNativeStack-dev.js:1686)```

## Environment

`-- react-native-calendars@1.5.8
`-- expo@19.0.1

**Comment by hkung77** [comment]
@joseguya 
I had this issue also, but some how I fixed it when I provided Locales for all locales (Even english). 
I only provided Locales for japanese, but when i added it for english it began to work..

```
   const locale = DeviceInfo.getDeviceLocale().slice(0, 2) === 'ja' ? 'ja' : 'en';
    LocaleConfig.defaultLocale = locale;
    moment.locale(locale);

    LocaleConfig.locales[locale] = {
      monthNamesShort: moment.months(),
      monrthNames: moment.months(),
      dayNames: moment.weekdays(),
      dayNamesShort: moment.weekdaysShort(),
    };
```

I hope this helps..

**Comment by tautvilas** [comment]
This has been resolved in 1.6.0
